src/Logger/EventLog.hs

```diff
@@ -22,79 +22,90 @@ import Data.String.Conversions
 import GHC.Generics (Generic)
 import Servant
 
+import qualified Data.Aeson as Aeson
 import qualified Data.Csv as CSV
 import qualified Data.Text as ST
+import qualified Generics.Generic.Aeson as Aeson
 import qualified Generics.SOP as SOP
 
 import Data.UriPath
 import Frontend.Path as U
 import Types
 
 
--- * event logs
+-- * types
 
-data EventLog = EventLog URL [EventLogItem]
-  deriving (Eq, Ord, Show, Read, Generic)
+data EventLog = EventLog URL [EventLogItemWarm]
+  deriving (Generic)
 
-data EventLogItem = EventLogItem IdeaSpace Timestamp User EventLogItemValue
-  deriving (Eq, Ord, Show, Read, Generic)
+data EventLogItem' user topic idea comment =
+    EventLogItem' IdeaSpace Timestamp user (EventLogItemValue' user topic idea comment)
+  deriving (Eq, Show, Generic)
 
-data URLEventLogItem = URLEventLogItem URL EventLogItem
+data EventLogItemValue' user topic idea comment =
```

Comment:
good idea.  that's an artefact of a previous state of the PR.  i'll take care of this.